### PR TITLE
Enhancement/use property and function name to register

### PR DESCRIPTION
**Pull request by piiertho**
This use property and function name as default registration name.

### CODE REVIEW

File: kt/godot-runtime/src/main/kotlin/godot/runtime/Registration.kt [hunk 1]
```diff
@@ -64,12 +64,14 @@ class ClassBuilderDsl<T : KtObject>(
...
-                          func: (T, P0) -> R,
-                          returnValueConverter: (R) -> KtVariant,
-                          p0Converter: (KtVariant) -> P0,
-                          arg: KtPropertyInfoBuilderDsl.() -> Unit,
-                          returns: KtPropertyInfoBuilderDsl.() -> Unit) {
+    fun <P0, R> function(
+            func: KFunction2<T, P0, R>,
+            returnValueConverter: (R) -> KtVariant,
+            p0Converter: (KtVariant) -> P0,
+            funcName: String = func.name.camelToSnakeCase(),
```
Comment (raniejade) [inline kt/godot-runtime/src/main/kotlin/godot/runtime/Registration.kt:71]: do we need it as a parameter? I don't we will ever need to pass in a different name

**Review by raniejade — CHANGES_REQUESTED** [review]

File: kt/godot-runtime/src/main/kotlin/godot/runtime/Registration.kt (hunk 1 above)
Comment (piiertho) [reply kt/godot-runtime/src/main/kotlin/godot/runtime/Registration.kt:71]: I wonder if someone wants to register it with a different name, if its method name conflicts with another from godot.
In this case I would say user have to change method name. But as I don't know all use cases, I just let the possibility to override it.

Comment (raniejade) [reply kt/godot-runtime/src/main/kotlin/godot/runtime/Registration.kt:71]: They have to change the name of their method then :P I don't that we should be the one to adjust for that.

Comment (piiertho) [reply kt/godot-runtime/src/main/kotlin/godot/runtime/Registration.kt:71]: I don't have strong opinion regarding this

**Review by raniejade — APPROVED** [review]

**Pull request merged**